packages/auto-dag-data/src/encryption/index.ts

```diff
@@ -0,0 +1,61 @@
+import { Crypto } from '@peculiar/webcrypto'
+import { asyncByChunk } from '../utils/async.js'
+import { EncryptorAlgorithm, EncryptorOptions, PasswordGenerationOptions } from './types.js'
+
+const crypto = new Crypto()
+
+export const ENCRYPTING_CHUNK_SIZE = 1024 * 1024
+const IV_SIZE = 16
+const TAG_SIZE = 16
+const ENCRYPTED_CHUNK_SIZE = ENCRYPTING_CHUNK_SIZE + IV_SIZE + TAG_SIZE
+
+export const getKeyFromPassword = async ({ password, salt }: PasswordGenerationOptions) => {
+  const encoder = new TextEncoder()
+  const saltHash =
+    typeof salt === 'string' ? await crypto.subtle.digest('SHA-256', encoder.encode(salt)) : salt
+
+  const passwordPlusSalt = Buffer.concat([Buffer.from(saltHash), encoder.encode(password)])
+  const passwordHash = await crypto.subtle.digest('SHA-256', passwordPlusSalt)
+
+  return await crypto.subtle.importKey('raw', passwordHash, { name: 'AES-GCM' }, false, [
```

Comment:
I think it would be better to use `deriveKey` here with with [PBKDF2](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/deriveKey#supported_algorithms). Something like:
```
const getKeyFromPassword = async ({ password, salt }: PasswordGenerationOptions) => {
  const encoder = new TextEncoder()
  const saltHash = typeof salt === 'string' 
    ? await crypto.subtle.digest('SHA-256', encoder.encode(salt)) 
    : salt

  // First import password as key material
  const keyMaterial = await crypto.subtle.importKey(
    'raw',
    encoder.encode(password),
    'PBKDF2',
    false,
    ['deriveBits', 'deriveKey']
  )

  // Then use PBKDF2 to derive the actual key
  return await crypto.subtle.deriveKey(
    {
      name: 'PBKDF2',
      salt: saltHash,
      iterations: 100000,
      hash: 'SHA-256'
    },
    keyMaterial,
    { name: 'AES-GCM', length: 256 },
    false,
    ['encrypt', 'decrypt']
  )
}
```

Reply:
Yeah, looks much better

Reply:
Aren't 100_000 iterations way too much?

Reply:
Nvm, I just look for it and looks like they aren't too much kind of surprising though